packages/story-editor/src/components/panels/document/taxonomies/HierarchicalTermSelector.js

```diff
@@ -254,49 +258,55 @@ function HierarchicalTermSelector({ noParentId = NO_PARENT_VALUE, taxonomy }) {
         label={taxonomy.labels.search_items}
         options={categories}
         onChange={handleClickCategory}
-        noOptionsText={taxonomy.labels.not_found}
+        noOptionsText={taxonomy.labels?.not_found}
       />
-      <LinkButton
-        ref={toggleRef}
-        aria-expanded={false}
-        onClick={handleToggleNewCategory}
-        $isVisible={showAddNewCategory}
-      >
-        {taxonomy.labels.add_new_item}
-      </LinkButton>
-      {showAddNewCategory ? (
-        <AddNewCategoryForm ref={formRef} onSubmit={handleSubmit}>
-          <Input
-            autoFocus
-            name={taxonomy.labels.new_item_name}
-            label={taxonomy.labels.new_item_name}
-            value={newCategoryName}
-            onChange={handleChangeNewCategoryName}
-            hasFocus={hasFocus}
-          />
-          <Label htmlFor={dropdownId}>{taxonomy.labels.parent_item}</Label>
-          <DropDown
-            id={dropdownId}
-            ariaLabel={taxonomy.labels.parent_item}
-            options={dropdownCategories}
-            selectedValue={selectedParent}
-            onMenuItemClick={handleParentSelect}
-          />
-          <ButtonContainer>
-            <AddNewCategoryButton
-              disabled={!newCategoryName.length}
-              type="submit"
-            >
-              {taxonomy.labels.add_new_item}
-            </AddNewCategoryButton>
-            <AddNewCategoryButton
-              aria-expanded
-              onClick={handleToggleNewCategory}
-            >
-              {__('Cancel', 'web-stories')}
-            </AddNewCategoryButton>
-          </ButtonContainer>
-        </AddNewCategoryForm>
+      {canCreateTerms ? (
+        <>
+          <LinkButton
+            ref={toggleRef}
+            aria-expanded={false}
+            onClick={handleToggleNewCategory}
+            $isVisible={showAddNewCategory}
+            id="expand_add_new_hierarchical_term"
```

Comment:
Is there a way to target this element in the e2e tests without a specific ID?

If not, why not use `data-testid`?

Reply:
good call, data ids will work just fine. Targeting without the id has been a little iffy, especially when trying to check it's not present for the contributor.